internal/cmd/launcher.go

```diff
@@ -0,0 +1,348 @@
+		return nil, errors.New("k6 cloud token is required when the Binary provisioning feature is enabled." +
+			" Set K6_CLOUD_TOKEN environment variable or execute the `k6 cloud login` command")
+	}
+
+	config := k6provider.Config{
+		BuildServiceURL:  gs.Flags.BuildServiceURL,
+		BuildServiceAuth: token,
+	}
+
+	provider, err := k6provider.NewProvider(config)
+	if err != nil {
+		return nil, err
+	}
+
+	binary, err := provider.GetBinary(gs.Ctx, deps)
+	if err != nil {
+		return nil, err
+	}
+
+	gs.Logger.
+		Info("A new k6 binary has been provisioned with version(s): ", formatDependencies(binary.Dependencies))
+
+	return &customBinary{binary.Path}, nil
+}
+
+func formatDependencies(deps map[string]string) string {
+	buffer := &bytes.Buffer{}
+	for dep, version := range deps {
+		buffer.WriteString(fmt.Sprintf("%s:%s ", dep, version))
+	}
+	return strings.Trim(buffer.String(), " ")
+}
+
+// extractToken gets the cloud token required to access the build service
+// from the environment or from the config file
+func extractToken(gs *state.GlobalState) (string, error) {
+	diskConfig, err := readDiskConfig(gs)
+	if err != nil {
+		return "", err
+	}
+
+	config, _, err := cloudapi.GetConsolidatedConfig(diskConfig.Collectors["cloud"], gs.Env, "", nil, nil)
+	if err != nil {
+		return "", err
+	}
+
+	return config.Token.String, nil
+}
+
+// analyze returns the dependencies for the command to be executed.
+// Presently, only the k6 input script or archive (if any) is passed to k6deps for scanning.
+// TODO: if k6 receives the input from stdin, it is not used for scanning because we don't know
+// if it is a script or an archive
+func analyze(gs *state.GlobalState, args []string) (k6deps.Dependencies, error) {
+	dopts := &k6deps.Options{
+		LookupEnv: func(key string) (string, bool) { v, ok := gs.Env[key]; return v, ok },
+		Manifest:  k6deps.Source{Ignore: true},
+	}
+
+	if !isScriptRequired(args) {
+		gs.Logger.
+			Debug("The command to execute does not require Binary provisioning")
+		return k6deps.Dependencies{}, nil
+	}
+
+	scriptname := scriptNameFromArgs(args)
+	if len(scriptname) == 0 {
+		gs.Logger.
+			Debug("The command did not receive an input script.")
+		return nil, errScriptNotFound
+	}
+
+	if scriptname == "-" {
+		gs.Logger.
+			Debug("Test script provided by Stdin is not yet supported from Binary provisioning feature.")
+		return nil, errUnsupportedFeature
+	}
+
+	if _, err := gs.FS.Stat(scriptname); err != nil {
+		gs.Logger.
+			WithField("path", scriptname).
+			WithError(err).
+			Debug("The requested test script's file is not available on the file system.")
+		return nil, errScriptNotFound
+	}
+
+	if strings.HasSuffix(scriptname, ".tar") {
+		dopts.Archive.Name = scriptname
+	} else {
+		dopts.Script.Name = scriptname
+	}
+
+	return k6deps.Analyze(dopts)
+}
+
+// isScriptRequired searches for the command and returns a boolean indicating if it is required to pass a script or not
+func isScriptRequired(args []string) bool {
```

Comment:
In case we run `k6 cloud run` with `--local-execution`, I think this should return `false`, as I guess it's equivalent to `k6 run`, isn't it? 🤔 
Or, do we intentionally want to run a custom binary locally in such case?

In any case, I'd suggest to add that case to `TestIsScriptRequired`, so we leave it explicit. Perhaps even accompanied with a code comment, as in my opinion isn't obvious.

Reply:
`could run --local-execution`  is expected to run with binary provisioning. We are not making a difference with `cloud run`.